### PR TITLE
UI/15406993 extensions with custom settings lazy consensus

### PR DESCRIPTION
**Pull request by krawczykkatarzyna**
prepare settings page and add logic to update custom settings parameters

Screen Shot 2023-06-06 at 01 18 44
Screen Shot 2023-06-06 at 01 19 30
Screen Shot 2023-06-06 at 01 30 58
Screen Shot 2023-06-06 at 01 20 04

### CODE REVIEW

File: src/components/frame/Extensions/Pages/LazyConsensusPage/LazyConsensusPage.tsx [hunk 1]
```diff
@@ -0,0 +1,101 @@
+import React from 'react';
```
Comment (KatarzynaStefanowiczPagepro) [inline src/components/frame/Extensions/Pages/LazyConsensusPage/LazyConsensusPage.tsx:1]: `src/components/frame/Extensions/pages/LazyConsensusPage/LazyConsensusPage.tsx`

Change the directory name to `pages` as we have on the main branch

File: src/components/common/Extensions/SpecificSidePanel/SpecificSidePanel.tsx [hunk 2]
```diff
@@ -9,18 +8,20 @@ import Version from './partials/Version';
 import ContractAddress from './partials/ContractAddress';
 import Developer from './partials/Developer';
 import styles from './SpecificSidePanel.module.css';
+import { useSpecificSidePanel } from './hooks';
 
 const displayName = 'common.Extensions.SpecificSidePanel';
 
-const SpecificSidePanel: FC<SpecificSidePanelProps> = ({ statuses, sidePanelData }) => {
+const SpecificSidePanel = () => {
```
Comment (KatarzynaStefanowiczPagepro) [inline src/components/common/Extensions/SpecificSidePanel/SpecificSidePanel.tsx:15]: `const SpecificSidePanel: FC = () => {`

File: src/components/common/Extensions/SpecificSidePanel/hooks.tsx [hunk 3]
```diff
@@ -0,0 +1,76 @@
+import React, { useMemo, useState } from 'react';
+import { useParams } from 'react-router-dom';
+import { format } from 'date-fns';
+import { useIntl } from 'react-intl';
+import { useExtensionData, useUserByNameOrAddress } from '~hooks';
```
Comment (KatarzynaStefanowiczPagepro) [inline src/components/common/Extensions/SpecificSidePanel/hooks.tsx:5]: Please add blank spaces between packages imports and components/tools

File: src/components/frame/Extensions/pages/LazyConsensusPage/LazyConsensusPage.tsx [hunk 4]
```diff
@@ -0,0 +1,101 @@
+import React, { FC } from 'react';
+import { useIntl } from 'react-intl';
+import { useLazyConsensusPage } from './hooks';
```
Comment (KatarzynaStefanowiczPagepro) [inline src/components/frame/Extensions/pages/LazyConsensusPage/LazyConsensusPage.tsx:3]: Please add blank spaces between packages imports and components/tools

File: src/components/frame/Extensions/pages/LazyConsensusPage/hooks.tsx [hunk 5]
```diff
@@ -0,0 +1,261 @@
...
+import { mapPayload, mergePayload, pipe } from '~utils/actions';
+import { mapExtensionActionPayload } from '~common/Extensions/ExtensionSetup/utils';
+import {
+  extensionContentSpeedOverSecurity,
+  extensionContentSecurityOverSpeed,
+  extensionContentTestingGovernance,
+} from './consts';
+import Toast from '~shared/Extensions/Toast/Toast';
+// import Toast from '~shared/Extensions/Toast/Toast';
+// import { toast } from 'react-toastify';
```
Comment (KatarzynaStefanowiczPagepro) [inline src/components/frame/Extensions/pages/LazyConsensusPage/hooks.tsx:23]: Remove unused imports

File: src/components/frame/Extensions/pages/LazyConsensusPage/hooks.tsx [hunk 6]
```diff
@@ -0,0 +1,261 @@
+import { useNavigate, useParams } from 'react-router-dom';
+import React, { useCallback, useLayoutEffect, useMemo, useState } from 'react';
+import { useIntl } from 'react-intl';
+import { useForm } from 'react-hook-form';
+import * as yup from 'yup';
+import { yupResolver } from '@hookform/resolvers/yup';
+import { Extension } from '@colony/colony-js/*';
+import { toast } from 'react-toastify';
+import { useAsyncFunction, useColonyContext, useExtensionData } from '~hooks';
```
Comment (KatarzynaStefanowiczPagepro) [inline src/components/frame/Extensions/pages/LazyConsensusPage/hooks.tsx:9]: Please add blank spaces between packages imports and components/tools

File: src/components/shared/Extensions/Accordion/partials/AccordionContent.tsx [hunk 7]
```diff
@@ -1,6 +1,8 @@
 import React, { FC } from 'react';
 import { AccordionItemProps } from '../types';
```
Comment (KatarzynaStefanowiczPagepro) [inline src/components/shared/Extensions/Accordion/partials/AccordionContent.tsx:2]: Please add blank spaces between packages imports and components/tools

File: src/components/shared/Extensions/Accordion/partials/AccordionHeader.tsx [hunk 8]
```diff
@@ -7,7 +7,7 @@ const displayName = 'Extensions.Accordion.partials.AccordionHeader';
 const AccordionHeader: FC<AccordionItemProps> = ({ title, isOpen }) => (
   <div>
     <div className="flex justify-between items-center pb-2 pt-4">
-      {title}
+      {title as React.ReactNode}
```
Comment (KatarzynaStefanowiczPagepro) [inline src/components/shared/Extensions/Accordion/partials/AccordionHeader.tsx:10]: You shouldn't use casting here.
Please set 
`title: string | ReactNode` in Props

Change everywhere where you use it.

File: src/components/shared/Extensions/Accordion/partials/ContentTypeText.tsx [hunk 9]
```diff
@@ -3,10 +3,10 @@ import { ContentTypeProps } from '../types';
 
 const displayName = 'Extensions.Accordion.partials.AccordionDetails';
 
-const ContentTypeText: FC<ContentTypeProps> = ({ title, subTitle }) => (
+const ContentTypeText: FC<ContentTypeProps> = ({ title = '', subTitle = '' }) => (
   <div>
-    <div className="font-medium text-gray-900 text-md pb-1">{title}</div>
-    <div className="text-sm font-normal text-gray-600">{subTitle}</div>
+    <div className="font-medium text-gray-900 text-md pb-1">{title as React.ReactNode}</div>
+    <div className="text-sm font-normal text-gray-600">{subTitle as React.ReactNode}</div>
```
Comment (KatarzynaStefanowiczPagepro) [inline src/components/shared/Extensions/Accordion/partials/ContentTypeText.tsx:9]: The same what with the 'title'

File: src/components/shared/Extensions/Accordion/types.ts [hunk 10]
```diff
@@ -1,19 +1,58 @@
...
+  details?: string;
 }
 
-export interface AccordionMocksContentDetails {
-  id: number;
+export interface AccordionContent extends ContentTypeProps {
+  paramName: string;
+  complementaryLabel: ComponentType;
+  id?: string;
+  description?: {
```
Comment (KatarzynaStefanowiczPagepro) [inline src/components/shared/Extensions/Accordion/types.ts:30]: `description:  MessageDescriptor | string`

File: src/components/shared/Extensions/Accordion/types.ts [hunk 11]
```diff
@@ -1,19 +1,58 @@
...
-  id: string;
-  title: string;
-  content: AccordionMocksContentDetails[];
+export interface ContentTypeProps {
+  title?:
+    | string
+    | {
+        defaultMessage: string;
+      };
+  subTitle?:
```
Comment (KatarzynaStefanowiczPagepro) [inline src/components/shared/Extensions/Accordion/types.ts:18]: `subTitle: string | React.Node`

File: src/components/shared/Extensions/Accordion/types.ts [hunk 12]
```diff
@@ -1,19 +1,58 @@
...
+  openIndex: number;
+  onOpenIndexChange: (newOpenIndex: number | undefined) => void;
 }
 
-export interface AccordionMocksContent {
-  id: string;
-  title: string;
-  content: AccordionMocksContentDetails[];
+export interface ContentTypeProps {
+  title?:
```
Comment (KatarzynaStefanowiczPagepro) [inline src/components/shared/Extensions/Accordion/types.ts:13]: `title: string | React.Node`

File: src/components/frame/Extensions/pages/LazyConsensusPage/hooks.tsx (hunk 5 above)
Comment (CzarekDryl) [inline src/components/frame/Extensions/pages/LazyConsensusPage/hooks.tsx]: Please remove comments

**Review by CzarekDryl — APPROVED** [review]

**Pull request merged**